kdc/krb5tgs.c

```diff
@@ -2229,6 +2215,20 @@ tgs_build_reply(astgs_request_t priv,
 	}
     }
 
+    /*
+     * Only add ticket signature if the requested server is not krbtgt, and
+     * either the header server is krbtgt or, in the case of renewal/validation
+     * if it was signed with PAC ticket signature and we verified it.
+     * Currently Heimdal only allow renewal of krbtgt anyway but that might
+     * change one day (see issue #763) so make sure to check for it.
+     */
+
+    if (kdc_issued &&
+	!krb5_principal_is_krbtgt(context, server->entry.principal))
+	add_ticket_sig = TRUE;
+
+    rodc_id = krbtgt_out->entry.kvno >> 16;
+
```

Comment:
Does this match the MS behaviour?   @josephsutton1 do you know?

Reply:
As it says, heimdal doesn't allow renewal of non-tgt but Windows does (can be tested with MIT's kinit -R, when the cache only contains a service ticket for instance). The aim is in the future when service-ticket renewal is made to work in Heimdal, that we match Windows behavior, which is afair; is the ticket had no PAC renew without PAC, if it had a PAC but no ticket-signature (like currently tickets issued by Windows via AS request, kinit -S host/xyz), renew the ticket and keep the PAC but do *not* add a ticket signature (the PAC can be unchanged), if it had ticket-signature then renew the ticket and regenerate all three signatures.

Reply:
Probably not relevant but may be interesting: Windows won't convert a RODC TGT to a full TGT if the renew flag is set. I noticed this during [TktBridgeAP](http://github.com/PADL/TktBridgeAP) development.